src/Ng/Jobs/LoopingNgJob.cs

```diff
@@ -38,7 +38,14 @@ protected LoopingNgJob(ILoggerFactory loggerFactory)
                 }
 
                 Logger.LogInformation("Running job.");
-                await RunInternal(cancellationToken);
+                try
+                {
+                    await RunInternal(cancellationToken);
+                }
+                catch(TransientException te)
+                {
+                    Logger.LogWarning("Transient exception: {0}. {1}", te.Message, te.InnerException.StackTrace??te.StackTrace);
```

Comment:
I think it's somewhat confusing that the job will still log "`Job finished!`" when it fails due to a transient exception. I would include that log within the `try` and add a "`Job failed!`" log to the `catch`. On a related note, it may be worth including a "`Job failed!`" log at [Program.cs#L73](https://github.com/NuGet/NuGet.Services.Metadata/blob/master/src/Ng/Program.cs#L73) as well.

E.g.
```
try
{
    await RunInternal(cancellationToken);
    Logger.LogInformation("Job finished!");
}
catch (TransientException te)
{
    Logger.LogWarning("Transient exception: {0}. {1}", te.Message, te.InnerException.StackTrace??te.StackTrace);
    Logger.LogWarning("Job failed!");
}
```